otx/algorithms/visual_prompting/adapters/pytorch_lightning/datasets/dataset.py

```diff
@@ -0,0 +1,332 @@
+"""Visual Prompting Dataset & DataModule."""
+
+# Copyright (C) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from typing import Dict, List, Optional, Union
+
+import cv2
+import numpy as np
+from omegaconf import DictConfig, ListConfig
+from torch import Tensor
+from torch.utils.data import DataLoader, Dataset
+from torchvision import transforms
+
+from otx.algorithms.common.utils.logger import get_logger
+from otx.algorithms.visual_prompting.adapters.pytorch_lightning.datasets.pipelines import (
+    MultipleInputsCompose,
+    Pad,
+    ResizeLongestSide,
+    collate_fn,
+)
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.image import Image
+from otx.api.entities.scored_label import ScoredLabel
+from otx.api.entities.shapes.polygon import Polygon
+from otx.api.entities.subset import Subset
+from otx.api.utils.shape_factory import ShapeFactory
+from pytorch_lightning import LightningDataModule
+
+logger = get_logger()
+
+
+class OTXVIsualPromptingDataset(Dataset):
+    """Visual Prompting Dataset Adaptor.
+
+    Args:
+        dataset (DatasetEntity): Dataset entity.
+        transform (MultipleInputsCompose): Transformations to apply to the dataset.
+        offset_bbox (int): Offset to apply to the bounding box, defaults to 0.
+    """
+
+    def __init__(self, dataset: DatasetEntity, transform: MultipleInputsCompose, offset_bbox: int = 0) -> None:
+
+        self.dataset = dataset
+        self.transform = transform
+        self.offset_bbox = offset_bbox
+
+        self.labels = dataset.get_labels()
+
+    def __len__(self) -> int:
+        """Get size of the dataset.
+
+        Returns:
+            int: Size of the dataset.
+        """
+        return len(self.dataset)
+
+    def __getitem__(self, index: int) -> Dict[str, Union[int, List, Tensor]]:
+        """Get dataset item.
+
+        Args:
+            index (int): Index of the dataset sample.
+
+        Returns:
+            Dict[str, Union[int, List, Tensor]]: Dataset item.
+        """
+        dataset_item = self.dataset[index]
+        item: Dict[str, Union[int, Tensor]] = {}
```

Comment:
```suggestion
        item: Dict[str, Union[int, Tensor]] = {"index": index}
```
how about to initialize at once by changing as above and removing a line below?

Reply:
Done.